sigridci/sigridci/sigrid_api_client.py

```diff
@@ -61,36 +61,55 @@ def callSigridAPI(self, path, body=None, contentType=None):
         else:
             return json.loads(responseBody)
 
-    def retry(self, operation, *, attempts=5, allow404=False, allowEmpty=True):
+    def retry(self, operation, *, attempts=5, allow404=False, allowEmpty=True, server="Sigrid"):
         for attempt in range(attempts):
             try:
                 response = operation()
                 if allowEmpty or response != {}:
                     return response
             except urllib.error.HTTPError as e:
-                if e.code == 401:
-                    UploadLog.log(f"You are not authenticated to Sigrid (HTTP status {e.code} for {e.url}), please check if your token is valid")
-                    sys.exit(1)
-                elif e.code == 403:
-                    UploadLog.log(f"You are not authorized to access Sigrid for this system (HTTP status {e.code} for {e.url})")
-                    sys.exit(1)
-                elif e.code == 410:
-                    if e.reason:
-                        UploadLog.log(f"{e.reason} (HTTP status {e.code} for {e.url})")
-                    else:
-                        UploadLog.log(f"The system no longer exists (HTTP status {e.code} for {e.url})")
-                    sys.exit(1)
-                elif allow404 and e.code == 404:
+                if e.code == 404 and allow404:
                     return False
+                self.handleError(e, server)
```

Comment:
Should be `SigridApiClient.handleError(…)`, since that's static. You could also make those extra methods not static, since you're never calling them in an exclusively static context anyway.

Reply:
Oops! I decided to keep it static, and call it the right way.